beacon_node/execution_layer/src/engines.rs

```diff
@@ -1,22 +1,26 @@
 //! Provides generic behaviour for multiple execution engines, specifically fallback behaviour.
 
 use crate::engine_api::{
-    Error as EngineApiError, ForkchoiceUpdatedResponse, PayloadAttributes, PayloadId,
+    EngineCapabilities, Error as EngineApiError, ForkchoiceUpdatedResponse, PayloadAttributes,
+    PayloadId,
 };
 use crate::HttpJsonRpc;
 use lru::LruCache;
 use slog::{debug, error, info, warn, Logger};
 use std::future::Future;
 use std::sync::Arc;
+use std::time::Duration;
 use task_executor::TaskExecutor;
 use tokio::sync::{watch, Mutex, RwLock};
 use tokio_stream::wrappers::WatchStream;
-use types::{Address, ExecutionBlockHash, Hash256};
+use types::ExecutionBlockHash;
 
 /// The number of payload IDs that will be stored for each `Engine`.
 ///
 /// Since the size of each value is small (~100 bytes) a large number is used for safety.
+/// FIXME: check this assumption now that the key includes entire payload attributes which now includes withdrawals
```

Comment:
```suggestion
/// Since the size of each value is small (~800 bytes) a large number is used for safety.
```

I think we can stick with a cache size of `512`. Here's my working:

- hash: `32`
- attributes:
  - timestamp: `8`
  - prev_randao: `32`
  - suggested_fee_recip: `20`
  - withdrawals: `44 * 16 = 704`
- payload_id: `8`

The size of each entry is `32+8+32+20+704+8 = 804` bytes.

The total cache would be `512 * 804 ~= 411KB` , which I think is very reasonable.